gadget/params.c

```diff
@@ -231,23 +235,23 @@ create_gadget_parameter_set()
     param_declare_int(ps, "SnapshotWithFOF", REQUIRED, 0, "Enable Friends-of-Friends halo finder.");
     param_declare_int(ps, "FOFSaveParticles", OPTIONAL, 1, "Save particles in the FOF catalog.");
     param_declare_double(ps, "FOFHaloLinkingLength", OPTIONAL, 0.2, "Linking length for Friends of Friends halos.");
-    param_declare_int(ps, "FOFHaloMinLength", OPTIONAL, 32, "");
+    param_declare_int(ps, "FOFHaloMinLength", OPTIONAL, 32, "Minimum number of particles per FOF Halo.");
     param_declare_double(ps, "MinFoFMassForNewSeed", OPTIONAL, 5e2, "Minimal Mass for seeding tracer particles ");
     param_declare_double(ps, "TimeBetweenSeedingSearch", OPTIONAL, 1e5, "Time Between Seeding Attempts: default to a a large value, meaning never.");
 
     /*Black holes*/
     param_declare_int(ps, "BlackHoleOn", REQUIRED, 1, "Enable Blackhole ");
-    param_declare_double(ps, "BlackHoleAccretionFactor", OPTIONAL, 100, "");
-    param_declare_double(ps, "BlackHoleEddingtonFactor", OPTIONAL, 3, "");
-    param_declare_double(ps, "SeedBlackHoleMass", OPTIONAL, 5e-5, "");
+    param_declare_double(ps, "BlackHoleAccretionFactor", OPTIONAL, 100, "BH accretion rate as a fraction of the Bondi accretion rate.");
+    param_declare_double(ps, "BlackHoleEddingtonFactor", OPTIONAL, 3, "Maximum Black hole accretion as a function of Eddington.");
+    param_declare_double(ps, "SeedBlackHoleMass", OPTIONAL, 5e-5, "Minimal black hole seed mass in internal mass units.");
 
-    param_declare_double(ps, "BlackHoleNgbFactor", OPTIONAL, 2, "");
+    param_declare_double(ps, "BlackHoleNgbFactor", OPTIONAL, 2, "Factor by which to increase the number of neighbours for a black hole.");
 
-    param_declare_double(ps, "BlackHoleMaxAccretionRadius", OPTIONAL, 99999., "");
-    param_declare_double(ps, "BlackHoleFeedbackFactor", OPTIONAL, 0.05, "");
-    param_declare_double(ps, "BlackHoleFeedbackRadius", OPTIONAL, 0, "");
+    param_declare_double(ps, "BlackHoleMaxAccretionRadius", OPTIONAL, 99999., "Maximum neighbour search radius for black holes.");
+    param_declare_double(ps, "BlackHoleFeedbackFactor", OPTIONAL, 0.05, " Fraction of the black hole luminosity to turn into thermal energy");
+    param_declare_double(ps, "BlackHoleFeedbackRadius", OPTIONAL, 0, "Maximum comoving (if > 0) radius at which the black hole feedback energy is deposited.");
 
-    param_declare_double(ps, "BlackHoleFeedbackRadiusMaxPhys", OPTIONAL, 0, "");
+    param_declare_double(ps, "BlackHoleFeedbackRadiusMaxPhys", OPTIONAL, 0, "Maximum physical (if > 0) radius at which the black hole feedback energy is deposited.");
```

Comment:
If set, the physical .... When both this flag and BlackHelFeedbackRadius are both set, the smaller radius is used.